Tests/t_ldap_dn.py

```diff
@@ -158,15 +158,42 @@ def test_dn2str(self):
                 [('dc', 'example', 1)],
                 [('dc', 'com', 1)]
             ]),
-            'uid=test\\, 42,ou=Testing,dc=example,dc=com'
+            'uid=test\\2C 42,ou=Testing,dc=example,dc=com'
         )
         self.assertEqual(
             ldap.dn.dn2str([
                 [('cn', 'äöüÄÖÜß', 4)],
                 [('dc', 'example', 1)],
                 [('dc', 'com', 1)]
             ]),
-            'cn=äöüÄÖÜß,dc=example,dc=com'
+            r'cn=\C3\A4\C3\B6\C3\BC\C3\84\C3\96\C3\9C\C3\9F,dc=example,dc=com'
```

Comment:
API change: I don't find a way to prevent this

Reply:
This is definitely intentional and the old value incorrect, we're telling libldap to treat the value as non-printable (`LDAP_AVA_NONPRINTABLE == 0x04`) and escape accordingly.

Reply:
ah yes, I overlooked that!

Reply:
Same vor `LDAP_AVA_STRING`: 
```python
>>> ldap.dn.dn2str([[('cn', 'äöüÄÖÜß', ldap.AVA_BINARY)],[('dc', 'example', 1)],[('dc', 'com', 1)]])
'cn=#C3A4C3B6C3BCC384C396C39CC39F,dc=example,dc=com'
>>> ldap.dn.dn2str([[('cn', 'äöüÄÖÜß', ldap.AVA_NULL)],[('dc', 'example', 1)],[('dc', 'com', 1)]])
'cn=\\C3\\A4\\C3\\B6\\C3\\BC\\C3\\84\\C3\\96\\C3\\9C\\C3\\9F,dc=example,dc=com'
>>> ldap.dn.dn2str([[('cn', 'äöüÄÖÜß', ldap.AVA_STRING)],[('dc', 'example', 1)],[('dc', 'com', 1)]])
'cn=\\C3\\A4\\C3\\B6\\C3\\BC\\C3\\84\\C3\\96\\C3\\9C\\C3\\9F,dc=example,dc=com'
>>> ldap.dn.dn2str([[('cn', 'äöüÄÖÜß', ldap.AVA_NONPRINTABLE)],[('dc', 'example', 1)],[('dc', 'com', 1)]])
'cn=\\C3\\A4\\C3\\B6\\C3\\BC\\C3\\84\\C3\\96\\C3\\9C\\C3\\9F,dc=example,dc=com'
```